### PR TITLE
Add npm-scripts

### PR DESCRIPTION
**Pull request by afontcu**
There are a lot of tweaks due to, well, TypeScript. I think it makes sense that we use nuxt-clean as a testing env, and formalize changes in `npm-scripts` once we've been playing with it a bit.

### CODE REVIEW

File: jest.setup.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import '@testing-library/jest-dom'
```
Comment (afontcu) [inline jest.setup.ts:1]: Jest setup file need to end in `.ts` for jest-dom to automatically pick up its types 😅

File: lint-staged.config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const merge = require('lodash.merge')
+const lintStagedConfig = require('@holaluz/npm-scripts').lintStaged
+
+delete lintStagedConfig['*.{js,vue}']
```
Comment (afontcu) [inline lint-staged.config.js:5]: nasty but comfortable XD

File: src/shared/Result.ts [hunk 3]
```diff
@@ -70,6 +70,7 @@ class Ok<T, E> {
 }
 
 class Err<T, E> {
+  // eslint-disable-next-line node/handle-callback-err
```
Comment (afontcu) [inline src/shared/Result.ts:73]: yeah, eslint-node is _that_ stupid

File: .eslintrc.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
-  root: true,
-  env: {
-    browser: true,
-    node: true,
-  },
+/* eslint-disable @typescript-eslint/no-var-requires */
+const merge = require('lodash.merge')
+const eslintConfig = require('@holaluz/npm-scripts').eslint
+
+delete eslintConfig.parserOptions
```
Comment (afontcu) [inline .eslintrc.js:5]: we can't override `parserOptions` with babel-eslint if we're using TS. This is the fastest solution :)

**Review by edufarre — APPROVED** [review]

**Pull request merged**